From 3d4225cae5462aeab81d50717b4cf6dab16b4354 Mon Sep 17 00:00:00 2001
From: Madalin Bucur <madalin.bucur@freescale.com>
Date: Thu, 8 Aug 2013 15:07:34 +0300
Subject: [PATCH 318/987] dpaa_eth: free bp platform device

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

The platform device was not released.

Signed-off-by: Madalin Bucur <madalin.bucur@freescale.com>
Change-Id: I44c134aba26f5eca5cad60fba3d0051ae9f99c8f
Reviewed-on: http://git.am.freescale.net:8181/3911
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Radulescu Ruxandra Ioana-B05472 <ruxandra.radulescu@freescale.com>
Reviewed-by: Rivera Jose-B46482 <Jose.G.Rivera@freescale.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
index c42cdb9..219de47 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
@@ -738,6 +738,9 @@ _dpa_bp_free(struct dpa_bp *dpa_bp)
 
 	dpa_bp_array[bp->bpid] = 0;
 	bman_free_pool(bp->pool);
+
+	if (bp->dev)
+		platform_device_unregister(to_platform_device(bp->dev));
 }
 
 void __cold __attribute__((nonnull))
-- 
1.9.1

